src/Uno.UI/UI/Xaml/UIElement.skia.cs

```diff
@@ -60,9 +60,31 @@ public bool UseLayoutRounding
 				typeof(UIElement),
 				new FrameworkPropertyMetadata(true, propertyChangedCallback: (o, _) => (o as IBorderInfoProvider)?.UpdateBorderThickness()));
 
+		/// <summary>
+		/// Represents the final calculated opacity of the element.
+		/// </summary>
+		/// <remarks>
+		/// This property should never be directly set, and its value should always be calculated through coercion (see <see cref="CoerceHitTestVisibility(DependencyObject, object, bool)"/>.
+		/// </remarks>
+		[GeneratedDependencyProperty(DefaultValue = 1.0, Options = FrameworkPropertyMetadataOptions.AffectsRender | FrameworkPropertyMetadataOptions.Inherits, CoerceCallback = true, ChangedCallback = false)]
```

Comment:
Not a fan of increasing coercion usages.

Reply:
As it's not a very common scenario, I'd keep a static `List<WeakReference>` of `ContentPresenter` that are used for native embedding, then, on *any* opacity update in the visual tree, calculate the opacity for things that are in the list. This is beneficial IMO because:

1. Avoids coercion, which ideally should be removed completely.
2. Moving the logic to where it really belongs, which is ContentPresenter. The CalculatedOpacity is really not relevant to UIElement, which also avoids increasing the size of `UIElement`s unnecessarily (even though the increase isn't a big deal)
3. Avoids the calculation of opacity in many cases where it's actually irrelevant/unused
4. Opacity isn't something that will change very often, esp with native embedding, so the extra cost of re-calculating the `List<WeakReference>` isn't a big deal.